models/migrations/v1_22/v292.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2024 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package v1_22 //nolint
+
+import (
+	"code.gitea.io/gitea/models/project"
+
+	"xorm.io/builder"
+	"xorm.io/xorm"
+)
+
+// CheckProjectColumnsConsistency ensures there is exactly one default board per project present
+func CheckProjectColumnsConsistency(x *xorm.Engine) error {
+	var projects []project.Project
+	if err := x.SQL("SELECT DISTINCT `p`.`id`, `p`.`creator_id` FROM `project` `p` WHERE (SELECT COUNT(*) FROM `project_board` `pb` WHERE `pb`.`project_id` = `p`.`id` AND `pb`.`default` = ?) != 1", true).
```

Comment:
I'd like to have pagination in this migration!

Reply:
Honestly I have no idea what you mean or how to do it 🙈

Reply:
Itterate over it in batches ... just look at older migrations

Reply:
Oh so the thing I did before 😆

Reply:
Uh you did?

Reply:
Yep, and then it was removed again.
Also, I don't quite see the benefit of batching here.
How does it help with anything?

Reply:
I guess the only reason to batch is when amount of project boards is really massive and the SQL operation would risk running into timeouts maybe.

Reply:
Done